.github/workflows/build_test.yml

```diff
@@ -0,0 +1,47 @@
+name: ROS2 CI
+
+on:
+  pull_request:
+    branches:
+      - 'develop'
+      - 'main'
+
+jobs:
+  test_environment:
+    runs-on: [ubuntu-latest]
+    strategy:
+      fail-fast: false
+      matrix:
+        ros_distribution:
+          - foxy
+          - rolling
+        include:
+          # Foxy Fitzroy (June 2020 - May 2023)
+          - docker_image: ubuntu:focal
+            ros_distribution: foxy
+            ros_version: 2
+          # Rolling Ridley  (June 2020 - Present)
+          - docker_image: ubuntu:focal
+            ros_distribution: rolling
+            ros_version: 2
+    container:
+      image: ${{ matrix.docker_image }}
+    steps:
+      - name: setup directories
+        run: mkdir -p ros_ws/src
+      - name: checkout
+        uses: actions/checkout@v2
+        with:
+          path: ros_ws/src
+      - name: setup ROS environment
+        uses: ros-tooling/setup-ros@master
```

Comment:
So, I'm not a big fan of using `setup-ros` because building the environment takes _so incredibly long_ relative to the tests that actually need to be run. Would you be OK with using their pre-built Docker images instead? e.g. `rostooling/setup-ros-docker:ubuntu-focal-ros-foxy-ros-base-latest` and `rostooling/setup-ros-docker:ubuntu-focal-ros-rolling-ros-base-latest`

Reply:
100% yes. i actually get frustrated with how long it takes too. i didnt know they provided a pre-built docker image too!

Reply:
You do lose the ability to verify your manifest and build scripts are correct I believe (as in: transitive dependencies might not get detected correctly).